File: common/public/src/main/java/dcp/mc/pstp/config/ProtectedEntities.java
package dcp.mc.pstp.config;

import org.jetbrains.annotations.NotNull;

public record ProtectedEntities(
        boolean tamable,
        boolean fox,
        boolean horses,
        boolean players,
        @NotNull String @NotNull [] blacklist,
        @NotNull String @NotNull [] whitelist
)
{
    ProtectedEntities() {
        this(true, true, true, false, new String[0], new String[0]);
    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/ProtectedEntities.kt
package dcp.mc.pstp.config
@JvmRecord
data class ProtectedEntities(
    val tamable: Boolean = true,
    val fox: Boolean = true,
    val horses: Boolean = true,
    val players: Boolean = false,
    val blacklist: List<String> = emptyList()
)